app/controllers/activity_forms_controller.rb

```diff
@@ -74,6 +77,12 @@ def update
       return
     end
 
+    if step == :implementing_organisation && @activity.errors.present?
```

Comment:
I'm trying to decide whether we want to have this behaviour covered in the controller tests - I noticed there aren't any feature tests for this new step in the form, is that because this is covered enough in the new `show` controller tests?

I think at the very least it would be good to have this step covered in `spec/features/users_can_create_a_third_party_project_level_activity_spec.rb` - what do you think?

Reply:
I'm trying to find the best balance between adding slow expensive tests and making sure all the paths are covered! This unhappy path in particular could only be reached if an org is made inactive while a user is in the process of assigning it as implem org, so a minimal amount of testing seems ideal.

Reply:
I think adding to the existing `spec/features/users_can_create_a_third_party_project_level_activity_spec.rb` to visit this page and set an implementing organisation (the happy path) is probably enough for me.

Reply:
Have added a test for this.